src/sentry/tagstore/legacy/backend.py

```diff
@@ -62,6 +62,13 @@ def setup_deletions(self, **kwargs):
 
         from sentry.deletions import default_manager as deletion_manager
         from sentry.deletions.base import ModelRelation, ModelDeletionTask
+        from sentry.models import Event
+
+        deletion_manager.add_bulk_dependencies(Event, [
+            lambda instance_list: ModelRelation(EventTag,
+                                                {'event_id__in': [i.id for i in instance_list]},
+                                                ModelDeletionTask),
+        ])
```

Comment:
Moved this into `legacy` because it's not restricted by `project_id`. I figure in `v2` we'll just let cleanup handle it?

Reply:
Why couldn't we do this within v2? All instances here would be within the same partition key.

So you can just do like, `partition_key={'project_id': instance_list[0].project_id}`

Reply:
Oh right, yeah I moved this before I got to the partition bit and :face_with_head_bandage: